### PR TITLE
ISPN-6748: Deprecate Singleton Mode for Persistent Stores.

### PR DESCRIPTION
**Pull request by ryanemerson**
https://issues.jboss.org/browse/ISPN-6748

### CODE REVIEW

**Comment by galderz** [comment]
@ryanemerson 46 new failures, they look related to the changes, can you please have a look? Thx :)

**Comment by ryanemerson** [comment]
@galderz I will take another look at this when I return from PTO.

**Comment by ryanemerson** [comment]
@galderz I have fixed the problems from before. The current test failures don't seem related to my changes (Hotrod).

**Comment by slaskawi** [comment]
There are still some tests failing: http://ci.infinispan.org/viewLog.html?buildId=40575&buildTypeId=bt9&guest=1

Retriggering CI.

File: core/src/main/java/org/infinispan/configuration/cache/SingletonStoreConfigurationBuilder.java [hunk 1]
```diff
@@ -11,6 +11,11 @@
 import org.infinispan.commons.configuration.attributes.AttributeSet;
 import org.infinispan.configuration.global.GlobalConfiguration;
 
+/**
+ * @deprecated Singleton writers will be removed in 9.1. If it is desirable that all nodes don't write to the underlying store
```
Comment (slaskawi) [inline core/src/main/java/org/infinispan/configuration/cache/SingletonStoreConfigurationBuilder.java]: If we deprecate them in 9.0, we probably won't be able to remove them until 10... Am I right @tristantarrant ?

**Comment by slaskawi** [comment]
The code looks fine (let's have another look at those CI failures as soon as TeamCity rebuilds this PR). The only concern that I have is removing it in `9.1`. I think we should do it in `10` (but maybe there is a good argument for that).

**Comment by tristantarrant** [comment]
I agree with @slaskawi , removal won't happen until 10

**Comment by ryanemerson** [comment]
@slaskawi @tristantarrant Version updated to 10.0. Can we get this in before the release tomorrow?

**Comment by ryanemerson** [comment]
@slaskawi @tristantarrant The issues with the tests were resolved with a pull --rebase. This should be good to merge.

**Comment by danberindei** [comment]
Integrated, thanks Ryan!

**Pull request closed without merging**